### PR TITLE
Development dbsettings

### PR DESCRIPTION
**Pull request by keatonwalker**
Added database configs to appsetting.json and updated readme.

Also eliminated postgres database env variables.

### CODE REVIEW

File: .kube/gke-deployment.yml [hunk 1]
```diff
@@ -53,21 +53,19 @@ spec:
...
-          value: db
-        - name: WEBAPI__DATABASE__PORT
-          value: '5432'
-        - name: WEBAPI__DATABASE__DB
-          value: webapi
-        - name: WEBAPI__DATABASE__USERNAME
-          value: postgres
-        - name: WEBAPI__DATABASE__PASSWORD
-          value: what password
         - name: WEBAPI__DATABASE__PEPPER
```
Comment (steveoh) [inline .kube/gke-deployment.yml]: let's move the pepper value as well so all of the webapi:database things are in the same place

File: src/api/appsettings.json [hunk 2]
```diff
@@ -16,6 +16,13 @@
...
       "weight": 1,
       "pathToLocator": "/arcgis/rest/services/Geolocators/"
-    }]
+    }],
+    "database": {
+      "host": "db",
+      "port": "5432",
+      "db": "webapi",
+      "username": "postgres",
+      "password": "what password"
```
Comment (steveoh) [inline src/api/appsettings.json]: this is missing the pepper

**Comment by keatonwalker** [comment]
> db might need to be in a different appsettings environment eg `appsettings.staging.json` because `db` will not resolve for local development. Right?
Yes for all the kube clusters it will need to be `db` which is the name of the database service.

**Comment by steveoh** [comment]
Do you want to revert that change and create the new `appsettings.production.json` file with the default docker settings?

**Comment by steveoh** [comment]
Do you want to revert that change and create the new files?

I suppose it should be 

`appsettings.json` with the defaults
`appsettings.development.json` with any development changes
`appsettings.production.json` with any changes for docker/production

**Comment by keatonwalker** [comment]
It is `ASPNETCORE_ENVIRONMENT` that will select the appsettings file correct?

**Comment by steveoh** [comment]
Yes, and I believe it mixes into the default `appsettings.json`.

https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-2.1#json-configuration-provider

**Comment by keatonwalker** [comment]
I think we won't want to commit `appsettings.production.json` because it now has the database password.

**Comment by steveoh** [comment]
Right, we should not commit a production file. We can commit a development file.

File: README.md [hunk 3]
```diff
@@ -217,7 +217,7 @@ Make sure to first change the `kubectl` context to the GKE cluster with `gcloud`
 
 app-config for mounting [appsettings.json](src/api/appsettings.json).
 
-- `kubectl create configmap app-config --from-file=src/api/appsettings.json`
+- `kubectl create configmap app-config --from-file=appsettings.json=./.kube/appsettings.kube.json`
```
Comment (keatonwalker) [inline README.md]: Creating the configmap this way allows us to mount other json files as any of the appsettings.*.json files

Comment (steveoh) [reply README.md]: oh ok

File: .kube/kube-deployment.yml [hunk 4]
```diff
@@ -66,7 +65,7 @@ spec:
         - name: ASPNETCORE_URLS
           value: http://+:1337
         - name: ASPNETCORE_ENVIRONMENT
-          value: Development
+          value: Developement
```
Comment (steveoh) [inline .kube/kube-deployment.yml]: this is spelled wrong

File: README.md [hunk 5]
```diff
@@ -41,7 +41,7 @@ Currently
 
 ## Configuration
 
-To make the project as flexible as possible, the connection strings, urls to services, etc required by the web API are read from environment variables at application startup. Environmental variables are the sole mechanism for storing application secrets in this project. They work well with [docker](docker-compose.override.yml), [k8s](kube-deployment.yml), and local development. The following will describe what is required before the application will function properly.
+To make the project as flexible as possible, the connection strings, urls to services, etc required by the web API are read from environment variables or [appsettings.json](src/api/appsettings.json) at application startup. Environmental variables will overide values in appsetttings. They both work well with [docker](docker-compose.override.yml), [k8s](.kube/kube-deployment.yml), and local development. The following will describe what is required before the application will function properly.
```
Comment (steveoh) [inline README.md]: let's remove the environment variable part all together and stick to one story.

**Pull request merged**